src/pages/guides/terraform/s3-replicate.mdx

```diff
@@ -0,0 +1,278 @@
+  byte_length = 8
+}
+
+resource "aws_iam_role" "replication" {
+  name               = "tf-iam-role-replication-${random_id.iam_role_id.hex}"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "aws_iam_policy_document" "replication" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetReplicationConfiguration",
+      "s3:ListBucket",
+    ]
+
+    resources = [aws_s3_bucket.bucket.arn]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObjectVersionForReplication",
+      "s3:GetObjectVersionAcl",
+      "s3:GetObjectVersionTagging",
+    ]
+
+    resources = ["${aws_s3_bucket.bucket.arn}/*"]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ReplicateObject",
+      "s3:ReplicateDelete",
+      "s3:ReplicateTags",
+    ]
+
+    resources = ["${aws_s3_bucket.destination.arn}/*"]
+  }
+}
+
+resource "aws_iam_policy" "replication" {
+  name   = "tf-iam-role-policy-replication-${random_id.iam_role_id.hex}"
+  policy = data.aws_iam_policy_document.replication.json
+}
+
+resource "aws_iam_role_policy_attachment" "replication" {
+  role       = aws_iam_role.replication.name
+  policy_arn = aws_iam_policy.replication.arn
+}
+```
+
+## Set up replication to the destination bucket
+
+Next, we set up the replication configuration for the original bucket in the module.
+
+<Note>
+  As per the documentation, this config includes a filter so that only objects
+  prefixed with "foo" will be replicated.
+</Note>
+
+```hcl {{ tag:'aws/deploytf/.nitric/modules/bucket/main.tf' }}
+resource "aws_s3_bucket_replication_configuration" "replication" {
+  depends_on = [
+    aws_s3_bucket_versioning.destination,
+    aws_s3_bucket_versioning.source
+  ]
+
+  role   = aws_iam_role.replication.arn
+  bucket = aws_s3_bucket.bucket.id
+
+  rule {
+    id = "s3-replication-rule-${random_id.bucket_id.hex}"
+
+    filter {
+      prefix = "foo"
+    }
+
+    status = "Enabled"
+
+    destination {
+      bucket        = aws_s3_bucket.destination.arn
+      storage_class = "STANDARD"
+    }
+
+    delete_marker_replication {
+      status = "Enabled" # or "Disabled" based on your requirements
+    }
+  }
+}
+```
+
+## Building and using your updated provider
+
+The Nitric project includes a make file that will build and install your provider as `nitric/awstf@0.0.1` by default.
+
+Navigate to `nitric/cloud/aws` and run `make install` to build and install the modified provider binary.
+
+```bash
+cd nitric/cloud/aws
+
+make install
+```
+
+The provider can then be used directly in your project's stack file as follows.
+
+```yaml
+# The nitric provider to use
+provider: nitric/awstf@0.0.1
+
+# The target aws region to deploy to
+region: us-east-2
+```
+
+<Note>
+  If you don't have a stack file use the `nitric stack new` to create one.
+</Note>
+
+You'll also need to enable `beta-providers` in your Nitric project by adding the following to your project's nitric.yaml file:
```

Comment:
```suggestion
Because the [Terraform providers](https://nitric.io/docs/reference/providers/terraform) are in preview, you'll also need to enable `beta-providers` in your Nitric project by adding the following to your project's nitric.yaml file:
```